tools-local/windows/tool-helper.ps1

```diff
@@ -0,0 +1,322 @@
+﻿# Provides helper functions.
+
+function Get-DefaultScriptsFolder
+{
+    [CmdletBinding()]
+    param(
+        [ValidateNotNullOrEmpty()]
+        [parameter(Mandatory=$true, Position=0)]
+        [string]$RepositoryRoot
+    )
+
+    return Join-Path "$RepositoryRoot" "tools-local\windows"
+}
+
+function Get-OperatingSystemArchitecture
+{
+    if ([System.Environment]::Is64BitOperatingSystem)
+    {
+        return "64"
+    }
+
+    return "32"
+}
+
+function Get-ToolVersionsConfig
+{
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory=$true, Position=0)]
+        [string]$ToolName,
+        [ValidateNotNullOrEmpty()]
+        [parameter(Mandatory=$true, Position=1)]
+        [string]$RepositoryRoot
+    )
+
+    $toolVersionsFilePath = Join-Path "$RepositoryRoot" ".toolversions"
```

Comment:
I hope you are planning to cache this in some way. Currently everytime Get-ToolVersionsConfig or transitiviely Get-ToolConfigValue is called you re-read the file from disk to parse the value. We should read the tools version file once.